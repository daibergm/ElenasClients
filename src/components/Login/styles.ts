import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  scroll: {
    width: '100%',
    paddingHorizontal: 15,
  },
  subContainer: {
    flex: 1,
  },
  buttonContainer: {
    marginTop: 30,
    paddingBottom: 15,
  },
});
