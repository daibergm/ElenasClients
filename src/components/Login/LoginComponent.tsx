import React from 'react';
import { View, ScrollView } from 'react-native';
import { Formik } from 'formik';
import * as yup from 'yup';
import { Text } from 'react-native-elements';

// @Assets
import styles from './styles';

// @Types
import { Login } from '../../types/';

// @Components
import { Button, Input } from '../common';

// @Constants
import { TEST_IDS, ERRORS } from '../../constants/';

// @Vars
const initialValues = {
  cellphone: '',
  password: '',
};

const validationSchema = yup.object().shape({
  cellphone: yup.string().trim().required('Este campo es obligatorio.'),
  password: yup.string().trim().required('Este campo es obligatorio.'),
});

interface Props {
  submitFunction: (values: Login) => void;
  loading: boolean;
  apiError?: string;
}

function LoginComponent({ submitFunction, loading, apiError }: Props) {
  const onSubmit = (values: Login) => submitFunction(values);

  return (
    <View style={styles.container}>
      <Formik
        initialValues={initialValues}
        validationSchema={validationSchema}
        onSubmit={onSubmit}>
        {({
          values,
          handleChange,
          errors,
          setFieldTouched,
          isValid,
          handleSubmit,
          touched,
          dirty,
        }) => (
          <ScrollView
            alwaysBounceVertical={false}
            showsVerticalScrollIndicator={false}
            style={styles.scroll}
            keyboardShouldPersistTaps="handled">
            <View style={styles.subContainer}>
              <Input
                testID={TEST_IDS.cellphoneInput}
                value={values.cellphone}
                placeholder="Celular"
                errorMessage={
                  touched.cellphone ? (errors.cellphone as string) : undefined
                }
                onChangeText={handleChange('cellphone')}
                onBlur={() => setFieldTouched('cellphone')}
              />
              <Input
                testID={TEST_IDS.passwordInput}
                value={values.password}
                placeholder="Contraseña"
                errorMessage={
                  touched.password ? (errors.password as string) : undefined
                }
                secureTextEntry
                onChangeText={handleChange('password')}
                onBlur={() => setFieldTouched('password')}
              />
              <View style={styles.buttonContainer}>
                <Button
                  title="INICIAR SESIÓN"
                  testID={TEST_IDS.submitButton}
                  disabled={!isValid || !dirty}
                  onPress={handleSubmit}
                  loading={loading}
                />
              </View>
              {apiError && (
                <View testID={TEST_IDS.errorLabel}>
                  <Text style={styles.errorLabel}>{ERRORS[apiError]}</Text>
                </View>
              )}
            </View>
          </ScrollView>
        )}
      </Formik>
    </View>
  );
}

export default LoginComponent;
