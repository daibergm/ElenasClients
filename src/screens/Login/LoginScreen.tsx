import React, { useContext } from 'react';
import { SafeAreaView } from 'react-native';

// @Context
import { AccountContext } from '../../context/';

// @types
import { Login } from '../../types/user';

// @Components
import { LoginComponent } from '../../components/';

// @Styles
import styles from './styles';

function LoginScreen() {
  const { onLogin, isLoading } = useContext(AccountContext);

  const onSubmit = async (data: Login) => onLogin && onLogin(data);

  return (
    <SafeAreaView style={styles.container}>
      <LoginComponent submitFunction={onSubmit} loading={!!isLoading} />
    </SafeAreaView>
  );
}

export default LoginScreen;
