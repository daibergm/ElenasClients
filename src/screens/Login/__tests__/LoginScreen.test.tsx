import React from 'react';
import { render } from '@testing-library/react-native';

// @Screens
import Login from '../LoginScreen';

describe('Login test suits', () => {
  it('Render correctly', () => {
    const login = render(<Login />);
    expect(login).toMatchSnapshot();
  });
});
