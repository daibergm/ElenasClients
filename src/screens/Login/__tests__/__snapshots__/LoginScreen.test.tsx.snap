// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test suits Render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <RCTScrollView
      alwaysBounceVertical={false}
      keyboardShouldPersistTaps="handled"
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "paddingHorizontal": 15,
          "width": "100%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "rgba(0, 173, 238, 0.65)",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "height": 48,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Celular"
                placeholderTextColor="rgba(0, 173, 238, 0.65)"
                style={
                  Object {
                    "color": "#00adee",
                    "flex": 1,
                    "fontSize": 16,
                    "minHeight": 40,
                  }
                }
                testID="cellphoneInput"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#dc3545",
                  "fontSize": 12,
                  "margin": 5,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomColor": "rgba(0, 173, 238, 0.65)",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "height": 48,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Contraseña"
                placeholderTextColor="rgba(0, 173, 238, 0.65)"
                secureTextEntry={true}
                style={
                  Object {
                    "color": "#00adee",
                    "flex": 1,
                    "fontSize": 16,
                    "minHeight": 40,
                  }
                }
                testID="passwordInput"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#dc3545",
                  "fontSize": 12,
                  "margin": 5,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "marginTop": 30,
                "paddingBottom": 15,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="submitButton"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "hsl(208, 8%, 90%)",
                      "borderColor": "#2089dc",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 40,
                      "justifyContent": "center",
                      "opacity": 0.65,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "hsl(208, 8%, 63%)",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    INICIAR SESIÓN
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
